pkg/api/volume/formatter.go

```diff
@@ -0,0 +1,91 @@
+package volume
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rancher/apiserver/pkg/apierror"
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/schemas/validation"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/harvester/harvester/pkg/api/vm"
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+)
+
+const (
+	actionExport = "export"
+)
+
+func Formatter(request *types.APIRequest, resource *types.RawResource) {
+	resource.Actions = make(map[string]string, 1)
+	resource.AddAction(request, actionExport)
+}
+
+type ExportActionHandler struct {
+	images v1beta1.VirtualMachineImageClient
+}
+
+func (h ExportActionHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	if err := h.do(rw, req); err != nil {
+		status := http.StatusInternalServerError
+		if e, ok := err.(*apierror.APIError); ok {
+			status = e.Code.Status
+		}
+		rw.WriteHeader(status)
+		_, _ = rw.Write([]byte(err.Error()))
+		return
+	}
+	rw.WriteHeader(http.StatusOK)
+	_, _ = rw.Write([]byte("Successfully export the volume."))
```

Comment:
@gitlawr  @guangbochen  I have to add this line (write sth, anything to http.ResponseWriter) to stop this "parsererror"
![Screenshot from 2021-09-22 23-11-40](https://user-images.githubusercontent.com/2396545/134465466-8cdfe789-1acd-44fb-ba0a-401316f85e2f.png)
: